### PR TITLE
update to protocols

### PR DESCRIPTION
**Pull request by jwagenaar**
# Description

Change location from which protocol links are fetched in the dataset details page.
https://www.wrike.com/open.htm?id=738446289

## Type of change
# How Has This Been Tested?
Locally tested:
1) Check that datasets with references use the references to protocols
2) Check that datasets without references use the records
3) Check that datasets without references only show protocols with DOI

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: pages/datasets/_datasetId.vue [hunk 1]
```diff
@@ -860,22 +860,56 @@ export default {
   },
   methods: {
     /**
-     * Returns protocol records in a dataset's model if they exist
+     * Sets active tab
+     * @param {String} activeLabel
+     */
+    setActiveTab: function(activeLabel) {
```
Comment (alan-wu) [inline pages/datasets/_datasetId.vue:866]: Is this function used anywhere?

File: pages/datasets/_datasetId.vue [hunk 2]
```diff
@@ -920,6 +954,9 @@ export default {
           })
       }
     },
+    linkFromDoi: function(doi) {
```
Comment (alan-wu) [inline pages/datasets/_datasetId.vue:957]: Is this used anywhere?

**Review by alan-wu — COMMENTED** [review]
There are two unused functions but everything else looks good to me.

File: pages/datasets/_datasetId.vue [hunk 3]
```diff
@@ -860,22 +860,56 @@ export default {
...
-          this.errorLoading = true
+      if (this.datasetInfo.externalPublications.length !== 0) {
+        const allPubs = this.datasetInfo.externalPublications
+        const allProtocols = allPubs.filter(
+          x => x.relationshipType === 'IsSupplementedBy'
+        )
+        this.datasetRecords = allProtocols.map(obj => {
+          let rObj = {}
+          rObj['url'] = `https://doi.org/` + obj.doi
+          return rObj
```
Comment (hsorby) [inline pages/datasets/_datasetId.vue:889]: I would return the object 
```
{url: `https://doi.org/${obj.doi}`}
```

File: pages/datasets/_datasetId.vue [hunk 4]
```diff
@@ -860,22 +860,56 @@ export default {
...
+            const records = propOr([], 'records', response)
+            if (records.length !== 0) {
+              // that means protocol records exist
+              const allProtocols = records.filter(protocol =>
+                protocol.properties.url.startsWith('https://doi.org')
+              )
+              this.datasetRecords = allProtocols.map(obj => {
+                let rObj = {}
+                rObj['url'] = obj.properties.url
+                return rObj
```
Comment (hsorby) [inline pages/datasets/_datasetId.vue:904]: Just return the object:
```
{url: obj.properties.url}
```

**Review by hsorby — CHANGES_REQUESTED** [review]

File: pages/datasets/_datasetId.vue (hunk 1 above)
Comment (jwagenaar) [reply pages/datasets/_datasetId.vue:866]: yes, it is used as a call-back on line 162 I think.

File: pages/datasets/_datasetId.vue (hunk 2 above)
Comment (jwagenaar) [reply pages/datasets/_datasetId.vue:957]: I don't think so. I'll remove

File: pages/datasets/_datasetId.vue (hunk 3 above)
Comment (jwagenaar) [reply pages/datasets/_datasetId.vue:889]: yes, that makes sense

**Review by alan-wu — APPROVED** [review]
Thanks, looks good to me.

**Pull request merged**